### PR TITLE
Avoid unneeded calculation in SeriesPeriodDetect

### PR DESCRIPTION
**Pull request by waynexia**
### Changelog category (leave one):
- Not For Changelog

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Avoid unneeded calculation in SeriesPeriodDetect.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [x]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

File: src/Functions/seriesPeriodDetectFFT.cpp [hunk 1]
```diff
@@ -153,12 +153,8 @@ class FunctionSeriesPeriodDetectFFT : public IFunction
             return true;
         }
 
-        std::vector<double> xfreq(spec_len);
         double step = 0.5 / (spec_len - 1);
-        for (size_t i = 0; i < spec_len; ++i)
-            xfreq[i] = i * step;
-
-        auto freq = xfreq[idx];
```
Comment (waynexia) [inline src/Functions/seriesPeriodDetectFFT.cpp:161]: `xfreq` is calculated for every frequency but only the `idx` one is used. This patch removes other unneeded calculations and use `idx * step` directly.

Comment (nickitat) [reply src/Functions/seriesPeriodDetectFFT.cpp:161]: very strange code, indeed

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit b8944abe0ec73ca386e851ce96d5f7ddaf3d254e with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/66320/b8944abe0ec73ca386e851ce96d5f7ddaf3d254e/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/66320/b8944abe0ec73ca386e851ce96d5f7ddaf3d254e/style_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Review by nickitat — APPROVED** [review]

**Comment by nickitat** [comment]
i slightly adjusted the PR's title and changelog entry if you don't mind.

**Comment by waynexia** [comment]
I'm not sure if the [CI failure](https://github.com/ClickHouse/ClickHouse/actions/runs/9873051468/job/27265858181?pr=66320#logs) is related to this patch. And I can't find the corresponding error output from CI log. Do you know where the error is @nickitat ? Thanks in advance.

**Comment by nickitat** [comment]
> I'm not sure if the [CI failure](https://github.com/ClickHouse/ClickHouse/actions/runs/9873051468/job/27265858181?pr=66320#logs) is related to this patch. And I can't find the corresponding error output from CI log. Do you know where the error is @nickitat ? Thanks in advance.

most likely worker is killed by OOM during linking

**Comment by waynexia** [comment]
Thank you @nickitat !

**Pull request merged**